doc/source/rllib/rllib-learner.rst

```diff
@@ -0,0 +1,459 @@
+        MultiAgentBatch)
+    from ray.rllib.evaluation.postprocessing import Postprocessing
+
+    DUMMY_BATCH = {
+        SampleBatch.OBS: np.array(
+            [[0.1, 0.2, 0.3, 0.4], [0.5, 0.6, 0.7, 0.8], [0.9, 1.0, 1.1, 1.2]],
+            dtype=np.float32,
+        ),
+        SampleBatch.NEXT_OBS: np.array(
+            [[0.1, 0.2, 0.3, 0.4], [0.5, 0.6, 0.7, 0.8], [0.9, 1.0, 1.1, 1.2]],
+            dtype=np.float32,
+        ),
+        SampleBatch.ACTIONS: np.array([0, 1, 1]),
+        SampleBatch.PREV_ACTIONS: np.array([0, 1, 1]),
+        SampleBatch.REWARDS: np.array([1.0, -1.0, 0.5], dtype=np.float32),
+        SampleBatch.PREV_REWARDS: np.array([1.0, -1.0, 0.5], dtype=np.float32),
+        SampleBatch.TERMINATEDS: np.array([False, False, True]),
+        SampleBatch.TRUNCATEDS: np.array([False, False, False]),
+        SampleBatch.VF_PREDS: np.array([0.5, 0.6, 0.7], dtype=np.float32),
+        SampleBatch.ACTION_DIST_INPUTS: np.array(
+            [[-2.0, 0.5], [-3.0, -0.3], [-0.1, 2.5]], dtype=np.float32
+        ),
+        SampleBatch.ACTION_LOGP: np.array([-0.5, -0.1, -0.2], dtype=np.float32),
+        SampleBatch.EPS_ID: np.array([0, 0, 0]),
+        SampleBatch.AGENT_INDEX: np.array([0, 0, 0]),
+        Postprocessing.ADVANTAGES: np.array([0.1, 0.2, 0.3], dtype=np.float32),
+        Postprocessing.VALUE_TARGETS: np.array([0.5, 0.6, 0.7], dtype=np.float32),
+    }
+    default_batch = SampleBatch(DUMMY_BATCH)
+    DUMMY_BATCH = default_batch.as_multi_agent()
+    ADDITIONAL_UPDATE_KWARGS = {"timestep": 0, "sampled_kl_values": {DEFAULT_POLICY_ID: 1e-4}}
+
+    learner.build() # needs to be called on the learner before calling any functions
+
+
+.. tab-set::
+        
+    .. tab-item:: Updating a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            # This is a blocking update
+            results = learner_group.update(DUMMY_BATCH)
+
+            # This is a non-blocking update. The results are returned in a future
+            # call to `async_update`
+            results = learner_group.async_update(DUMMY_BATCH)
+
+            # This is an additional non-gradient based update.
+            learner_group.additional_update(**ADDITIONAL_UPDATE_KWARGS)
+
+        When updating a `LearnerGroup` you can perform blocking or async updates on batches of data. Async updates are necessary for implementing async algorithms such as APPO/IMPALA.
+        You can perform non-gradient based updates using `additional_update`.
+
+    .. tab-item:: Updating a Learner
+
+        .. testcode::
+            :skipif: True
+
+            # This is a blocking update.
+            result = learner.update(DUMMY_BATCH)
+
+            # This is an additional non-gradient based update.
+            learner_group.additional_update(**ADDITIONAL_UPDATE_KWARGS)
+
+        When updating a `Learner` you can only perform blocking updates on batches of data.
+        You can perform non-gradient based updates using `additional_update`.
```

Comment:
For update, async_update and async_update for additional_update, can we link to API refs?

Reply:
done.